File: src/main/java/org/afterlike/lucid/task/DelayedTask.java
package org.afterlike.lucid.task;

final class DelayedTask {
	private int ticks;
	private final Runnable runnable;
	DelayedTask(final int ticks, final Runnable runnable) {
		this.ticks = ticks;
		this.runnable = runnable;
	}

	boolean tick() {
		return ticks-- <= 0;
	}

	void run() {
		runnable.run();
	}
}

File: src/main/kotlin/org/afterlike/lucid/core/type/DelayedTask.kt
package org.afterlike.lucid.core.type

import org.afterlike.lucid.core.event.game.GameTickEvent

class DelayedTask(private var ticks: Int, private val runnable: () -> Unit) {

    fun tick(event: GameTickEvent): Boolean {
        if (event.phase != EventPhase.PRE) return false

        if (ticks-- <= 0) {
            runnable()
            return true
        }

        return false
    }
}